lib/src/http/common/request.dart

```diff
@@ -155,6 +155,7 @@ abstract class CommonRequest extends Object
   /// the [encoding].
   @override
   MediaType get contentType {
+    _contentType ??= defaultContentType;
```

Comment:
good catch - I had started updating this method but realized the logic below wasn't as simple as I remembered.. I removed the rest of the changes but not this line